crowbar_framework/app/models/api/upgrade.rb

```diff
@@ -213,6 +216,18 @@ def upgrade_controller_nodes
         # FIXME: save the global status information that this node is being upgraded
         node_api.upgrade
 
+        # Explicitly mark node1 as the cluster founder
+        # this is one time action only (for each cluster)
+        Api::Pacemaker.set_node_as_founder drbd_slave
+
+        # Remove "pre-upgrade" attribute from node1
+        # We must do it from the node where pacemaker is running
+        master_node_api = Api::Node.new drbd_master
+        master_node_api.remove_pre_upgrade_role_for drbd_slave
+
+        # FIXME: this should be one time action only (for each cluster)
+        master_node_api.delete_pacemaker_resources
```

Comment:
Again, need to check the return value.  (Or raise an exception on failure which gets caught.)